tezos/new_context/src/tezedge_context.rs

```diff
@@ -297,28 +304,27 @@ impl TezedgeIndex {
         };
 
         // first get node at key
-        let child_node_id = match storage.get_tree_node_id(root, first) {
-            Some(hash) => hash,
+        let child_node_id = match storage.tree_find_node(root, first) {
+            Some(node_id) => node_id,
             None => {
                 return Ok(Tree::empty());
             }
         };
 
         // get entry (from working tree)
         let child_node = storage.get_node(child_node_id)?;
-        if let Some(entry) = child_node.get_entry() {
-            match entry {
-                Entry::Tree(tree) => {
-                    return self.find_raw_tree(tree, &key[1..], storage);
-                }
-                Entry::Blob(_) => return Ok(Tree::empty()),
-                Entry::Commit { .. } => {
-                    return Err(MerkleError::FoundUnexpectedStructure {
-                        sought: "Tree/Blob".to_string(),
-                        found: "commit".to_string(),
-                    })
-                }
+        match child_node.get_entry() {
+            Some(Entry::Tree(tree)) => {
+                return self.find_raw_tree(tree, &key[1..], storage);
+            }
+            Some(Entry::Blob(_)) => return Ok(Tree::empty()),
+            Some(Entry::Commit { .. }) => {
+                return Err(MerkleError::FoundUnexpectedStructure {
+                    sought: "Tree/Blob".to_string(),
+                    found: "commit".to_string(),
+                })
             }
+            None => {}
```

Comment:
@sebastiencs can you add some comments in the code explaining this situation here? (it is similar to the issues I had when reworking `find_tree` I think, it is kind of unexpected to have the node but not being able to get the entry).

Reply:
@sebastiencs once we get into the cleanup work, lets use more descriptive names for these methods, like `get_entry_from_working_tree` or something like that.